kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -74,12 +74,17 @@ object Clock:
 
     object Deadline:
         /* WARNING: Low-level API meant for integrations, libraries, and performance-sensitive code. See AllowUnsafe for more details. */
-        class Unsafe(end: Instant, clock: Clock.Unsafe):
+        class Unsafe(endInstant: Maybe[Instant], clock: Clock.Unsafe):
```

Comment:
I noticed that the deadline impl wasn't considering `Duration.Infinity` correctly.